### PR TITLE
cppmangle: Follow up on #10021

### PR DESCRIPTION
**Pull request by Geod24**
Best reviewed commits by commit.
As I was adding more tests, I also realized `version(linux)` should be `version(Posix)` for `compilable/cppmangle.d`, so changed that too.

CC @SSoulaimane

### CODE REVIEW

File: src/dmd/cppmangle.d [hunk 1]
```diff
@@ -665,6 +665,8 @@ private final class CppMangleVisitor : Visitor
         //printf("prefix_name(%s)\n", s.toChars());
         if (substitute(s))
             return;
+        if (isStd(s))
+            return buf.writestring("St");
```
Comment (thewilsonator) [inline src/dmd/cppmangle.d]: This looks like it should be part of the last commit.

**Comment by thewilsonator** [comment]
Otherwise LGTM.

File: src/dmd/cppmangle.d (hunk 1 above)
Comment (Geod24) [reply src/dmd/cppmangle.d]: Fixed

**Review by thewilsonator — APPROVED** [review]

**Review by SSoulaimane — APPROVED** [review]
Good.

**Pull request merged**